### PR TITLE
text.details comparison

### PR DESCRIPTION
**Pull request by DonAurelio**

### CODE REVIEW

File: test/gen_source.rb [hunk 1]
```diff
@@ -22,11 +22,13 @@
   }
 
   btx_source_status_t btx_push_usr_messages(common_data_t *common_data, void *usr_data) {
+      <%- if data -%>
```
Comment (TApplencourt) [inline test/gen_source.rb:25]: you should not need the `if data`, we should inalizate data to `[]`

File: test/gen_source.rb [hunk 2]
```diff
@@ -47,29 +49,30 @@ def sanitize_value(field_value, field_class)
...
-
-      i = line.index('{')
-      head, _, tail = line.partition(': {')
-
-      field_values =  tail.scan(REGEXT_PRETTY).flatten
-      data = {
-        name: head.gsub(/[^0-9A-Za-z-]/, '_'), # Should reuse metababel mangling
-        field_values: field_values.zip(field_classes).map { |fvalue, fclass| sanitize_value(fvalue, fclass) }
-      }
+  if input_path
```
Comment (TApplencourt) [inline test/gen_source.rb:56]: we we call this function if we don't have an `input path`?

File: test/gen_source.rb [hunk 3]
```diff
@@ -82,8 +85,9 @@ def render_and_save(data, output_path)
   Usage: example.rb [options]
 
   Example:
-    ruby example.rb -t yaml -i stream_classes.yaml -o callbacks.c
-    ruby example.rb -t log -f EE -i babeltrace.log -o  callbacks.c
+    ruby example.rb -y stream_classes.yaml -i btx_log.txt -o callbacks.c
```
Comment (TApplencourt) [inline test/gen_source.rb:88]: please use `$0`

File: test/test_stream_classes_model.rb [hunk 4]
```diff
@@ -4,7 +4,7 @@ class TestSourceNoCommonField < Test::Unit::TestCase
   include TestSourceBase
   extend VariableAccessor
   include VariableClassAccessor
-
+
```
Comment (TApplencourt) [inline test/test_stream_classes_model.rb:7]: Please don't add space

File: test/test_stream_classes_model.rb [hunk 5]
```diff
@@ -87,3 +87,71 @@ def self.startup
     }
   end
 end
+
+class TestSourceDetailsComparisonAllTypes < Test::Unit::TestCase
```
Comment (TApplencourt) [inline test/test_stream_classes_model.rb:91]: I think we can refractor this code to use the `module` trick for having a general tests

File: test/test_cases_stream_classes_model/6.btx_model.yaml [hunk 6]
```diff
@@ -0,0 +1,99 @@
+---
+:stream_classes:
+- :name: sc1
+  :event_common_context_field_class:
+    :type: structure
+    :members:
+    - :name: cf_1
+      :field_class:
+        :type: bool
+        :cast_type: bt_bool
```
Comment (TApplencourt) [inline test/test_cases_stream_classes_model/6.btx_model.yaml:10]: Please don't cast the `bt_bool`, it's by default `bt_bool`

File: test/test_cases_stream_classes_model/6.btx_model.yaml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    - :name: cf_2
+      :field_class:
+        :type: integer_unsigned
+        :field_value_range: 64
+        :cast_type: uint64_t
```
Comment (TApplencourt) [inline test/test_cases_stream_classes_model/6.btx_model.yaml:15]: For text detail, I dont think we want to use `cast_type`. We never send message

File: test/test_stream_classes_model.rb (hunk 4 above)
Comment (DonAurelio) [reply test/test_stream_classes_model.rb:7]: Do you mean remove the space between the include and the first method definition?

From this 

```ruby
  include VariableClassAccessor
  
  def self.startup
```
To this 

```ruby
  include VariableClassAccessor
  def self.startup
```

File: test/test_stream_classes_model.rb (hunk 5 above)
Comment (DonAurelio) [reply test/test_stream_classes_model.rb:91]: Now, I can not imagine how, but lets discuss that, I think you can enlighten me !!

File: test/test_stream_classes_model.rb (hunk 4 above)
Comment (TApplencourt) [reply test/test_stream_classes_model.rb:7]: No you just added some space is this line. This is what the `diff` show

**Pull request merged**